### PR TITLE
feat: Courses not found component

### LINKED ISSUE #310: Improve "Unable to find saved courses" component
If users do not have any courses saved when attempting to register or look at their booklist, the message: "Unable to find saved courses for <Term>" appears. This can be somewhat vague, given users might not know "saved courses" are what appear on their timetables. Furthermore, we are just leaving users on their own to go add these.

The message should be re-worded to be more clear and provide a link/button to the timetable view for users to click and add courses.

TODO:
- [ ] Increase clarity of sentence
- [ ] Create link/button to timetable view
- [ ] make this it's own component since it is reused in different places
- [ ] write unit tests

Current view:

### PR DESCRIPTION
**Pull request by liamshatzel**
# Description

Created courses not found component to replace courses not found on registration and booklist page. 

Closes #310 

### Before

Screen Shot 2022-02-11 at 9 50 41 PM

### After

Screen Shot 2022-02-11 at 9 50 08 PM

## Checklist

- [ ] The code follows all style guidelines.
- [ ] The code passes all required tests.
- [ ] The code is documented.
- [x] The code includes tests.
- [x] I have self-reviewed my changes and have done QA.

## General Comments

Still in draft form, as it still doesn't pass my tests.

### CODE REVIEW

File: src/common/coursesnotfound/__tests__/CoursesNotFound.test.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { MemoryRouter } from 'react-router';
+
+import { renderWithRouter } from 'lib/utils/jest/index';
+import { getCurrentTerm, getReadableTerm } from 'lib/utils/terms';
+
+import { CoursesNotFound } from '../CoursesNotFound';
+
+const term = getCurrentTerm();
+it('is in the document', () => {
+  renderWithRouter(<CoursesNotFound />, { route: `/${getCurrentTerm()}` });
```
Comment (aomi) [inline src/common/coursesnotfound/__tests__/CoursesNotFound.test.tsx:13]: The reason why this doesn't work is that when you render this component, you end up having no `term` route parameter encapsulating it. 

Normally when you use the `useParams` and such, its within a `Route` component with the route params defined with something like `/:term` (look in `routes.tsx`). If you want to test this with the term showing up as expected, you need to wrap it with a `Route` component.

Alternatively you make this component not reliant on `useParams` and inject it into the page as props.

File: src/common/coursesnotfound/CoursesNotFound.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+              <Button
+                colorScheme="blue"
+                width="auto"
+                leftIcon={<HiOutlineCalendar />}
+                as={Link}
+                to={`/scheduler/${term}`}
+              >
+                {`${getReadableTerm(term)}`} timetable
+              </Button>
+            </Box>
```
Comment (keithradford) [inline src/common/coursesnotfound/CoursesNotFound.tsx:34]:
```suggestion
            {`Unable to find ${item} for `}
            <Text as="span" color={mode('black', 'white')}>
              {getReadableTerm(term)}
            </Text>
            <Box padding="4">
              {timetableButton && <Button
                colorScheme="blue"
                width="auto"
                leftIcon={<HiOutlineCalendar />}
                as={Link}
                to={`/scheduler/${term}`}
              >
                {`${getReadableTerm(term)}`} timetable
              </Button>}
            </Box>
```

**Comment by keithradford** [comment]
@liamshatzel As much as I want to keep this componentized, I think the context changes too much page to page, and we should favour clarity over simplicity here. Overall I think these should be the following cases:

- On `Explore courses` when no sections are found
    - `No sections offered for <term>`
- On registration
    -  `Unable to find saved courses from your <term> timetable` and then the button
- On _no textbooks found but courses saved_
    - `No textbooks found for your saved courses in <term>`
- On textbook page when no saved courses
    -  `Unable to find saved courses from your <term> timetable` and then the button

Thanks for all the work so far and sorry to backpedal on you after writing the tests and all that, was a lack of planning on my part!

**Comment by liamshatzel** [comment]
@keithradford For the textbook case is there a way can I check whether there are no registered courses vs no textbooks found?

**Pull request closed without merging**